docs/raw/push-notification-server.md

```diff
@@ -0,0 +1,672 @@
+# 16/PUSH-NOTIFICATION-SERVER
+
+> Version: 0.1
+>
+> Status: Raw
+>
+> Authors: Andrea Maria Piana <andreap@status.im>
+
+- [Push Notification Server](#16push-notification-server)
+  - [Reason](#reason)
+  - [Requirements](#requirements)
+  - [Registering with the push notification service](#registering-with-the-push-notification-server)
+  - [Re-registering with the push notification service](#re-registering-with-the-push-notification-server)
+  - [Changing options](#changing-options)
+  - [Unregistering from push notifications](#unregistering-from-push-notifications)
+  - [Advertising a push notification server](#advertising-a-push-notification-server)
+  - [Discovering a push notification server](#discovering-a-push-notificatino-server)
+  - [Querying the push notification service](#querying-the-push-notification-server)
+  - [Sending a push notification](#sending-a-push-notification)
+  - [Flow](#flow)
+    - [Registration process](#registration-process)
+    - [Sending a notification](#sending-a-notification)
+    - [Receiving a notification](#receiving-a-notification)
+  - [Protobuf description](#protobuf-description)
+    - [PushNotificationRegister](#pushnotificationregister)
+    - [PushNotificationPreferences](#pushnotificationpreferences)
+    - [PushNotificationDeviceToken](#pushnotificationdevicetoken)
+    - [PushNotificationOptions](#pushnotificationoptions)
+    - [PushNotificationPreferencesAcknowledgment](#pushnotificationpreferencesacknowledgement)
+    - [ContactCodeAdvertisement](#contactcodeadvertisement)
+    - [PushNotificationAdvertisementInfo](#pushnotificationadvertisementinfo)
+    - [PushNotificationQuery](#pushnotificationquery)
+    - [PushNotificationQueryInfo](#pushnotificationqueryinfo)
+    - [PushNotificationQueryResponse](#pushnotificationqueryresponse)
+    - [PushNotification](#pushnotification)
+    - [PushNotificationRequest](#pushnotificationrequest)
+    - [PushNotificationAcknowledgement](#pushnotificationacknowledgement)
+  - [Partitioned topic](#partitioned-topic)
+  - [FAQ](#faq)
+  - [Changelog](#changelog)
+    - [Version 0.1](#version-01)
+  - [Copyright](#copyright)
+
+## Reason
+
+Push notifications for IOS devices and for some android device can only be
+implemented relying on [APN service](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/APNSOverview.html#//apple_ref/doc/uid/TP40008194-CH8-SW1) for IOS or [Firebase](https://firebase.google.com/) for 
+Android devices that do not support background services or that often kill the 
+background service.
+Therefore Status provides a set of Push notification services that can be used
+to achieve this functionality.
+
+Because this can't be safely implemented in a privacy preserving manner, clients
+MUST be given an option not to enable push notifications and not to send push notifications.
+
+## Requirements
+
+Status needs to run their own [gorush](https://github.com/appleboy/gorush) 
```

Comment:
Second round: the party releasing the app. Which could be Status Gmbh (or PTE LTD?), or some other party.